### ISSUE TITLE
Segfault on ly on my system

### ISSUE THREAD

**Issue by ultra-azu** [fairyglade/ly#138]
Hello! I hope you have a good day. For some reason, according to my logs, ly is having a segmentation fault when started by xdm, in OpenRC. 

Here's the exact error:

     Oct 27 07:51:33 UltraArch-PC /etc/init.d/xdm[2817]: start-stop-daemon: failed to start `/usr/bin/ly'
     Oct 27 07:51:33 UltraArch-PC kernel: ly[2818]: segfault at 10 ip 000055b5e6b4ec10 sp 00007ffdff326b70 error 4 in ly[55b5e6b49000+9000]
     Oct 27 07:51:33 UltraArch-PC kernel: Code: 7e 00 00 83 f8 ff 74 0b 8b 05 b8 7e 00 00 83 f8 ff 75 29 83 7d fc ff 74 23
     83 7d f8 ff 74 1d 48 8b 05 dc 0a 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d 83 07 01 00 e8 1c 1e 00 00 8b 05 80 7e
     Oct 27 07:51:33 UltraArch-PC /etc/init.d/xdm[2819]:          ERROR: could not start the Display Manager

I'll happily provide any info needed if requested.

EDIT: BTW, 

System=Artix Linux(I know my hostname is misleading)
CPU=Ryzen 3 1200
GPU=GTX 1060 
RAM=8GB

This probably doesn't relate to this, but more info doesn't hurt.

**Comment by madstk1** [comment]
Hello,

I'm having a similar issue on Void Linux. I'm currently using this service-file:
```bash
#!/bin/sh
sv start agetty-tty1
exec /usr/bin/ly
```
Though, the result is similar:
```
[  319.941342] ly[2488]: segfault at 10 ip 000055bc1fb3fa18 sp 00007ffdc3d90220 error 4 in ly[55bc1fb39000+a000]
[  319.941354] Code: 81 00 00 83 f8 ff 74 0b 8b 05 1c 81 00 00 83 f8 ff 75 29 83 7d fc ff 74 23 83 7d f8 ff 74 1d 48 8b 05 54 0d 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d fb 09 01 00 e8 91 1e 00 00 8b 05 e4 80
[  320.946237] ly[2497]: segfault at 10 ip 000055a6feef3a18 sp 00007fffeea0fc50 error 4 in ly[55a6feeed000+a000]
[  320.946248] Code: 81 00 00 83 f8 ff 74 0b 8b 05 1c 81 00 00 83 f8 ff 75 29 83 7d fc ff 74 23 83 7d f8 ff 74 1d 48 8b 05 54 0d 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d fb 09 01 00 e8 91 1e 00 00 8b 05 e4 80
[  321.951692] ly[2503]: segfault at 10 ip 000055979f1fea18 sp 00007ffc9f6b6980 error 4 in ly[55979f1f8000+a000]
[  321.951704] Code: 81 00 00 83 f8 ff 74 0b 8b 05 1c 81 00 00 83 f8 ff 75 29 83 7d fc ff 74 23 83 7d f8 ff 74 1d 48 8b 05 54 0d 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d fb 09 01 00 e8 91 1e 00 00 8b 05 e4 80
[  322.955300] ly[2513]: segfault at 10 ip 000056085cda2a18 sp 00007ffd7bb7f8b0 error 4 in ly[56085cd9c000+a000]
[  322.955309] Code: 81 00 00 83 f8 ff 74 0b 8b 05 1c 81 00 00 83 f8 ff 75 29 83 7d fc ff 74 23 83 7d f8 ff 74 1d 48 8b 05 54 0d 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d fb 09 01 00 e8 91 1e 00 00 8b 05 e4 80
[  323.960206] ly[2518]: segfault at 10 ip 00005579e7812a18 sp 00007ffe714c3580 error 4 in ly[5579e780c000+a000]
[  323.960213] Code: 81 00 00 83 f8 ff 74 0b 8b 05 1c 81 00 00 83 f8 ff 75 29 83 7d fc ff 74 23 83 7d f8 ff 74 1d 48 8b 05 54 0d 01 00 48 83 c0 10 <48> 8b 00 48 89 c6 48 8d 3d fb 09 01 00 e8 91 1e 00 00 8b 05 e4 80
```

**Distro:** Void Linux
**Kernel:** Linux 5.3.13_1
**Init:** runit-2.1.2_10 (latest)

**Comment by jmi2k** [comment]
Similar results for me.

*Distro:* Void Linux
*Kernel:* 5.4.23_1
*Init:* 2.1.2_11

**Comment by FatmanUK** [comment]
Same segfault for me on Void.

Distro: Void
Kernel: 4.19.58_1
Init: runit-2.1.2_11

My service file:
```
#!/bin/sh
exec ly | logger -t ly
```

**Comment by jackghg** [comment]
Same here, artix runit. (issues 190, 203)

**Comment by GrbavaCigla** [comment]
Same issue here

**Comment by BerkeleyTrue** [comment]
Sam here:

Distro: Manjaro Linux
Kernal: 5.14.21-2
ininit: systemd 250.3-4

**Comment by thinkoid** [comment]
See the comment I left on #221

**Comment by AnErrupTion** [comment]
Assuming this is fixed (like #221), closing.
